### PR TITLE
Hb 04

### PR DESCRIPTION
**Pull request by Rudiachenko**

### CODE REVIEW

File: src/main/java/com/dev/cinema/dao/impl/CinemaHallDaoImpl.java [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+            }
+        }
+    }
+
+    @Override
+    public List<CinemaHall> getAll() {
+        try (Session session = HibernateUtil.getSessionFactory().openSession()) {
+            Query<CinemaHall> getAllMoviesQuery =
+                    session.createQuery("from CinemaHall", CinemaHall.class);
+            return getAllMoviesQuery.getResultList();
```
Comment (Nokafan) [inline src/main/java/com/dev/cinema/dao/impl/CinemaHallDaoImpl.java]: You can make code list shorter by using return session.createQuery("from CinemaHall", CinemaHall.class).getResultList();

File: src/main/java/com/dev/cinema/dao/impl/MovieDaoImpl.java [hunk 2]
```diff
@@ -12,27 +12,30 @@
...
             if (session != null) {
                 session.close();
             }
         }
     }
 
+    @Override
     public List<Movie> getAll() {
         try (Session session = HibernateUtil.getSessionFactory().openSession()) {
             Query<Movie> getAllMoviesQuery = session.createQuery("from Movie", Movie.class);
```
Comment (Nokafan) [inline src/main/java/com/dev/cinema/dao/impl/MovieDaoImpl.java]: Could be shorter :)

File: src/main/java/com/dev/cinema/dao/impl/MovieSessionDaoImpl.java [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+                session.close();
+            }
+        }
+    }
+
+    @Override
+    public List<MovieSession> findAvailableSessions(Long movieId, LocalDate date) {
+        try (Session session = HibernateUtil.getSessionFactory().openSession()) {
+            Query<MovieSession> query =
+                    session.createQuery("from MovieSession where movie_id = :movieId "
```
Comment (Nokafan) [inline src/main/java/com/dev/cinema/dao/impl/MovieSessionDaoImpl.java]: Good practice to write SQL reserved words in upperCase

**Review by Nokafan — COMMENTED** [review]
Good job.

File: src/main/java/com/dev/cinema/dao/impl/CinemaHallDaoImpl.java [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+            transaction = session.beginTransaction();
+            session.persist(cinemaHall);
+            transaction.commit();
+            return cinemaHall;
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new DataProcessingException("Can't insert cinema hall "
+                    + cinemaHall.getDescription(), e);
```
Comment (Sofasmile) [inline src/main/java/com/dev/cinema/dao/impl/CinemaHallDaoImpl.java]: Можна й так))
```suggestion
                    + cinemaHall.toString(), e);
```

File: src/main/java/com/dev/cinema/dao/impl/MovieSessionDaoImpl.java [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+    public List<MovieSession> findAvailableSessions(Long movieId, LocalDate date) {
+        try (Session session = HibernateUtil.getSessionFactory().openSession()) {
+            Query<MovieSession> query =
+                    session.createQuery("FROM MovieSession WHERE movie_id = :movieId "
+                    + "AND show_time BETWEEN :start AND :end", MovieSession.class);
+            LocalDateTime start = date.atTime(LocalTime.MIN);
+            LocalDateTime end = date.atTime(LocalTime.MAX);
+            query.setParameter("movieId", movieId);
+            query.setParameter("start", start);
+            query.setParameter("end", end);
```
Comment (Sofasmile) [inline src/main/java/com/dev/cinema/dao/impl/MovieSessionDaoImpl.java]:
```suggestion
            query.setParameter("start", date.atTime(LocalTime.MIN));
            query.setParameter("end", date.atTime(LocalTime.MAX));
```

File: src/main/java/com/dev/cinema/main/Main.java [hunk 6]
```diff
@@ -1,20 +1,70 @@
 package com.dev.cinema.main;
```
Comment (Sofasmile) [inline src/main/java/com/dev/cinema/main/Main.java]:
```suggestion
package com.dev.cinema;
```

**Review by Sofasmile — COMMENTED** [review]
good job

**Review by Sofasmile — APPROVED** [review]

**Pull request merged**